src/apps/replication/exe/config-zk.ini

```diff
@@ -151,6 +151,38 @@ partition_count = 1
 max_replica_count = 3
 
 [replication]
+; Notes on the failure detector:
+; 1. We mainly send beacon when checking the beacons. So in normal cases,
+;    the beacon_interval_seconds is useless if beacon_interval_seconds < check_interval_seconds.
+;    So you'd better set the beacon_interval_seconds larger than check_interval_seconds
+;
+; 2. Worker may disconnect from master in the period earlier that the lease_seconds
+;    to ensure the perfect FD. In the worst case, workers may lease "check_interval_seconds"
+;    earlier.
+;    While on the master side, it may claim a worker disconnected longer than the grace_seconds, which
+;    can also be "check_interval_seconds" later in the worst case.
+;
+; 3. In practice, your should set check_interval_seconds a small value for a fine-grained FD.
+;    For client, you may set it 1 second as it usually connect to a small number of masters.
+;    For master, you may set it 2 or 3 seconds.
+;
+; 4. the scale between beacon_interval and lease_periods is up to the rpc channel somewhat.
+;    In tcp, the channel itself supports the "keepalive" so you don't need to send beacons frequently.
+;    But in udp, you may want to try several times before the worker expires.
+;    So in tcp, the beacon_interval/lease_periods could be 10/30. And in UDP, 4/30 is acceptable
+;
+; 5. The lease_periods should be less than the grace_periods to tolerant the
+;    clock drift or delay by the OS scheduler.
```

Comment:
People may pay more attention to this if these comments are in source codes. So I move it to failure_detector.h, and resend a pr.
And the "should" is fixed, tks :). Please add more notes if necessary.